torch/_torch_docs.py

```diff
@@ -5787,6 +5787,90 @@ def merge_dicts(*dicts):
     tensor([  2.,   4.,   8.,  16.])
 """.format(**common_args))
 
+add_docstr(torch.float_power,
+           r"""
+float_power(input, exponent, *, out=None) -> Tensor
+
+Takes the power of each element in :attr:`input` with :attr:`exponent` and
+returns a tensor in float64 with the result.
+
+:attr:`exponent` can be either a single ``float`` number or a `Tensor`
+with the same number of elements as :attr:`input`.
+
+When :attr:`exponent` is a scalar value, the operation applied is:
+
+.. math::
+    \text{out}_i = x_i ^ \text{exponent}
+
+When :attr:`exponent` is a tensor, the operation applied is:
+
+.. math::
+    \text{out}_i = x_i ^ {\text{exponent}_i}
+""" + r"""
+When :attr:`exponent` is a tensor, the shapes of :attr:`input`
+and :attr:`exponent` must be :ref:`broadcastable <broadcasting-semantics>`.
+
+.. note:: This differs from the power function :func:`torch.pow` in that integers, float16, and float32 are
+          promoted to floats with a minimum precision of float64 so that the result is always inexact.
+          The intent is that the function will return a usable result for negative powers and
+          seldom overflow for positive powers.
+
+Args:
+    {input}
+    exponent (float or tensor): the exponent value
+
+Keyword args:
+    {out}
+
+Example::
+
+    >>> a = torch.randint(10, (4,))
+    >>> a
+    tensor([6, 4, 7, 1])
+    >>> torch.float_power(a, 2)
+    tensor([36., 16., 49.,  1.], dtype=torch.float64)
+
+    >>> a = torch.arange(1, 5)
+    >>> a
+    tensor([ 1,  2,  3,  4])
+    >>> exp = torch.arange(1, 5)
+    >>> exp
+    tensor([ 1,  2,  3,  4])
+    >>> torch.float_power(a, exp)
+    tensor([   1.,    4.,   27.,  256.], dtype=torch.float64)
+
+.. function:: float_power(self, exponent, *, out=None) -> Tensor
```

Comment:
Let's not document this twice. Our current docs do this but we should try to avoid it. I think everything below here can be cut.

Reply:
Addressed!